src/EnergyPlus/GeneralRoutines.cc

```diff
@@ -2139,4 +2139,106 @@ void TestReturnAirPathIntegrity(EnergyPlusData &state, IOFiles &ioFiles, bool &E
+    //    cp_moistair_average = Psychrometrics::PsyCpAirFnW(0.5 * (W2 + W1));
+    //    cp_moistair_average = cp_dryair + 0.5 * (W2 + W1) * cp_watervapor
+    // Q_latent = Q_total - Q_latent;
+    // or Q_latent = m_dot * hg_average * (W2 - W1);
+    //    hg_average = Psychrometrics::PsyHfgAvgFnTdb2Tdb1(TDB2, TDB1);
+
+    using Psychrometrics::PsyDeltaHSenFnTdb2W2Tdb1W1;
+    using Psychrometrics::PsyHFnTdbW;
+
+    TotalOutput = 0.0;
+    LatentOutput = 0.0;
+    SensibleOutput = 0.0;
+    if (MassFlow > 0.0) {
+        TotalOutput = MassFlow * (Psychrometrics::PsyHFnTdbW(TDB2, dW2) - Psychrometrics::PsyHFnTdbW(TDB1, dW1)); // total addition/removal rate, {W};
+        SensibleOutput = MassFlow * Psychrometrics::PsyDeltaHSenFnTdb2W2Tdb1W1(TDB2, dW2, TDB1, dW1); // sensible addition/removal rate, {W};
+        LatentOutput = TotalOutput - SensibleOutput;                                                  // latent addition/removal rate, {W}
+    }
+}
+
+void CalcZoneSensibleLatentOutput(Real64 const MassFlow,  // air mass flow rate, {kg/s}
+                                  Real64 const TDBEquip,  // dry-bulb temperature at equipment outlet {C}
+                                  Real64 const dWEquip,   // humidity ratio at equipment outlet
+                                  Real64 const TDBZone,   // dry-bulb temperature at zone air node {C}
+                                  Real64 const dWZone,    // humidity ratio at zone air node
+                                  Real64 &SensibleOutput, // sensible output rate (state 2 -> State 1), {W}
+                                  Real64 &LatentOutput,   // latent output rate (state 2 -> State 1), {W}
+                                  Real64 &TotalOutput     // total = sensible + latent putput rate (state 2 -> State 1), {W}
+)
+{
+
+    // Purpose:
+    // returns total, sensible and latent heat rate of transfer between equipment outlet
+    // and zone air node. The mosit energy transfer can be cooling and heating depending
+    // on the zone equipment outlet and zone air node conditions.
+
+    // Methodology:
+    // Q_total = m_dot * (hEquip - hZone)
+    // Q_sensible = m_dot * Psychrometrics::PsyDeltaHSenFnTdb2W2Tdb1W1(TDBEquip, dWEquip, TDBZone, dWZone);
+    // or Q_sensible = m_dot * cp_moistair_minHumRat * (TDBEquip - TDBZone)
+    // Q_latent = Q_total - Q_latent;
+
+    using Psychrometrics::PsyDeltaHSenFnTdbEquipTdbWZone;
+    using Psychrometrics::PsyHFnTdbW;
+
+    TotalOutput = 0.0;
+    LatentOutput = 0.0;
+    SensibleOutput = 0.0;
+    if (MassFlow > 0.0) {
+        TotalOutput = MassFlow * (Psychrometrics::PsyHFnTdbW(TDBEquip, dWEquip) -
+                                  Psychrometrics::PsyHFnTdbW(TDBZone, dWZone)); // total addition/removal rate, {W};
+        SensibleOutput =
+            MassFlow * Psychrometrics::PsyDeltaHSenFnTdb2W2Tdb1W1(TDBEquip, dWEquip, TDBZone, dWZone); // sensible addition/removal rate, {W};
```

Comment:
My concern for NOT using  PsyDeltaHSenFnTdbEquipTdbWZone() in CalcZoneSensibleLatentOutput() is that the later function's use varies a lot. Sometimes it is used to calculate total/sensible/latent outputs and other times it is used to calculate sensible/latent outputs only.

Reply:
PsyDeltaHSenFnTdbEquipTdbWZone() uses the zone HumRat to calculate sensible heat transfer rate and this may not be the right approach for CalcZoneSensibleLatentOutput() since we are providing supply and zone HumRat values.  If we decide to use PsyDeltaHSenFnTdbEquipTdbWZone() in CalcZoneSensibleLatentOutput() then the supply HumRat argument will not be used and the zone HumRat may not be the minimum. As the result, the latent component can be incorrect.

Reply:
@Nigusse For a given supply and zone condition, `CalcZoneSensibleLatentOutput` and `CalcZoneSensibleOutput` must return the same sensible load. The intent of these functions is to calculate the load impact on the zone from an air stream entering the zone, and if I recall correctly, we decided that the zone humrat was the appropriate one to use (even if it is not the minimum). Perhaps there are more use cases than just coil (change across the coil) and zone?

Reply:
@mjwitte Okay I will make the changes in CalcZoneSensibleLatentOutput() function as we did in the CalcZoneSensibleOutput() function. It is good to document why we are using ZoneHumRat in certain places and MinHumRat in other places.